src/internet_identity/src/anchor_management/tentative_device_registration.rs

```diff
@@ -109,8 +105,6 @@ fn get_verified_device(
     anchor_number: AnchorNumber,
     user_verification_code: DeviceVerificationCode,
 ) -> Result<DeviceData, VerifyTentativeDeviceResponse> {
-    trap_if_not_authenticated(&state::anchor(anchor_number));
```

Comment:
just to be clear, this was hoisted to `verify_tentative_device`, correct?

Reply:
Yes, exactly. This was the most ill-placed authentication check that I moved.